include/cantera/numerics/Integrator.h

```diff
@@ -175,10 +175,12 @@ class Integrator
         warn("setMethodType");
     }
 
-    //! Set the linear iterator.
-    virtual void setIterator(IterType t) {
-        warn("setInterator");
-    }
+    #if CT_SUNDIALS_VERSION < 40
+        //! Set the linear iterator.
+        virtual void setIterator(IterType t) {
+            warn("setInterator");
+        }
+    #endif
```

Comment:
Given that we never use any method besides Newton iteration, my suggestion would be to:
* deprecate `setIterator` and remove the `m_iter` variable
* remove the existing calls to `setIterator`, and
* directly call `CVodeCreate` with the `CV_NEWTON` constant if we're using an older version of Sundials.